### PR TITLE
Fallback to internal searcher if not in translog

### PR DESCRIPTION
**Pull request by pxsalehi**
This might not be a final fix for this. Eventually, we need to properly tie cleaning up 
of translog to live version map archive clean up (and unmprotobale shards updated).

### CODE REVIEW

**Comment by pxsalehi** [comment]
@elasticmachine update branch

File: server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java [hunk 1]
```diff
@@ -878,6 +878,10 @@ protected GetResult realtimeGetUnderLock(
         }
         boolean getFromSearcherIfNotInTranslog = getFromSearcher;
         if (versionValue != null) {
+            // It is possible that once we've seen the ID in the version map and attempt handling the get,
+            // we won't see the doc in the translog since it might have been moved out. In these cases, we
+            // should always fall back to get the doc from the internal searcher.
```
Comment (henningandersen) [inline server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java:883]: Can we perhaps note in the comment that ideally we should keep the translog around?

**Review by henningandersen — APPROVED** [review]
LGTM.

**Comment by pxsalehi** [comment]
@elasticmachine run elasticsearch-ci/part-2

(failure was https://github.com/elastic/elasticsearch/issues/99778)

File: server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java (hunk 1 above)
Comment (ywangd) [reply server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java:883]: Can we also add a comment to say this is also for the case when the location is not tracked in the versionMap which is the original use case for this flag? Basically this flag now cover two cases.

**Review by ywangd — APPROVED** [review]
LGTM

Pooya and I synced and he explain the thought process of this change. It was super helpful! Thanks!

**Comment by pxsalehi** [comment]
@elasticmachine run elasticsearch-ci/part-2

**Comment by pxsalehi** [comment]
@elasticmachine run elasticsearch-ci/part-2

(failure was https://github.com/elastic/elasticsearch/issues/100069)

**Comment by pxsalehi** [comment]
@elasticmachine run elasticsearch-ci/part-2

(failure was https://github.com/elastic/elasticsearch/issues/100081

**Comment by pxsalehi** [comment]
hmmm... can't really merge this due to unrelated CI failures, mostly esql related. Let's try one last time.
@elasticmachine run elasticsearch-ci/part-2

**Comment by pxsalehi** [comment]
@elasticmachine update branch

**Comment by pxsalehi** [comment]
@elasticmachine run elasticsearch-ci/part-2
(infra issue)

**Pull request merged**